### PR TITLE
Swap client ids in queue

### PR DESCRIPTION
**Pull request by Eunovo**
### Description

This turned out to be much harder than expected. I'm still testing this with sample app and working out bugs.

##### Checklist

- [x] `npm test` passes
- [x] `npm run build` works
- [x] tests are included

### CODE REVIEW

File: packages/datastore/datastore/src/replication/mutations/QueueUpdateProcessor.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+import { MutationRequest } from "./MutationRequest";
+import { LocalStorage, CRUDEvents } from "../../storage";
+import { Model } from "../../Model";
+import traverse from "traverse";
+
+/**
+ * Swaps all client ids for server ids in queue
+ * and updates corresponding item in LocalStorage
+ */
+export class QueueUpdateProcessor {
```
Comment (wtrocki) [inline packages/datastore/datastore/src/replication/mutations/QueueUpdateProcessor.ts:11]: I think we will need to also handle addingNew item to queue from here to being able to add mutexes (separate PR/issue). Thx.

File: packages/datastore/datastore/src/replication/mutations/QueueUpdateProcessor.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
+import { ModelMap } from "./MutationsQueue";
+import { MutationRequest } from "./MutationRequest";
+import { LocalStorage, CRUDEvents } from "../../storage";
+import { Model } from "../../Model";
+import traverse from "traverse";
```
Comment (wtrocki) [inline packages/datastore/datastore/src/replication/mutations/QueueUpdateProcessor.ts:5]: why traverse is needed?

**Review by wtrocki — APPROVED** [review]

File: packages/datastore/datastore/src/replication/mutations/QueueUpdateProcessor.ts (hunk 2 above)
Comment (Eunovo) [reply packages/datastore/datastore/src/replication/mutations/QueueUpdateProcessor.ts:5]: It allows us to swap the client IDs in place anywhere in the queue items array

**Comment by Eunovo** [comment]
@wtrocki this work still has bugs. I'm testing this in sample app

**Comment by wtrocki** [comment]
No problem. I see them but want to work of the recent version when testing other storage as well

**Pull request merged**